iwyu.cc

```diff
@@ -3975,14 +3975,18 @@ class IwyuAstConsumer
     if (!Base::TraverseTypedefDecl(decl))
       return false;
 
-    return HandleAliasedClassMethods(decl);
+    return current_ast_node()->in_forward_declare_context()
```

Comment:
I think it would be nicer to move this check into `HandleAliasedClassMethods`, e.g.
```
  bool HandleAliasedClassMethods(TypedefNameDecl* decl) {
    if (CanIgnoreCurrentASTNode())  return true;
    if (current_ast_node()->in_forward_declare_context())
      return true;
    ...
```

Reply:
I surely saw the duplication, but I thought that such bypasses should not be in `HandleAliasedClassMethods` function. Maybe it should be renamed into `HandleAliasedClassMethodsIfNeeded` or something else?

Reply:
`CanIgnoreCurrentASTNode` also potentially short-circuits `HandleAliasedClassMethods`, so I wouldn't worry about it too much. It makes sense to me for it to silently pass if it doesn't need to do anything.